drivers/src/idevid_csr.rs

```diff
@@ -0,0 +1,49 @@
+// Licensed under the Apache-2.0 license
+
+use core::mem::size_of;
+use zerocopy::{AsBytes, FromBytes};
+use zeroize::Zeroize;
+
+use crate::memory_layout;
+
+pub const MAX_CSR_SIZE: usize = 512;
+
+#[derive(FromBytes, AsBytes, Zeroize)]
+#[repr(C)]
+pub struct IDevIDCsr {
+    pub csr: [u8; MAX_CSR_SIZE],
```

Comment:
If we're going to have a get function, should the fields be non-public?

Reply:
Refactored this, but I will need to remove some more log statements to re-coop the bytes.